### PR TITLE
Use c++11 instead of boost, bump version to 1.0.0

### PR DESCRIPTION
**Pull request by scpeters**
The groundwork was laid to ease migration from boost to c++11, and this pull request implements the transition for `urdfdom_headers`. A separate pull request will be required for `urdfdom`. A quick summary:

* Use and export c++11 compiler flag
* Use `std::` pointers instead of `boost::`
* Add a custom replacement for `boost::split` (called `urdf::split_string`)
* Replace `boost::lexical_cast` calls with `std::sto[df]` and catch the corresponding exceptions

Due to these large changes, the major version is bumped to 1. We should create `0.4` branch from master before merging this. Also, note that due to the use of [SameMajorVersion](https://github.com/ros/urdfdom_headers/blob/master/CMakeLists.txt#L42) when writing our cmake version config file, urdfdom will have to be updated to explicitly find this version.

I have an [unboost branch in scpeters/urdfdom](https://github.com/scpeters/urdfdom/commits/unboost), but it's not quite ready. The only thing remaining is switching from boost test to gtest.

### CODE REVIEW

File: urdf_model/include/urdf_model/utils.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*********************************************************************
+* Software License Agreement (BSD License)
+*
+*  Copyright (c) 2008, Willow Garage, Inc.
```
Comment (jslee02) [inline urdf_model/include/urdf_model/utils.h]: Out of the curiosity, shouldn't we use the current year for licenses of new files?

Comment (scpeters) [inline urdf_model/include/urdf_model/utils.h]: Willow Garage doesn't exist either. I don't know what to do about licensing for new files, so I have just copied the old license.

I'll cc @tfoote and @gerkey for some other opinions.

Comment (tfoote) [inline urdf_model/include/urdf_model/utils.h]: New files get a new copyright(owner), by the author with the current date. It's typically recommended to use the same license for consistency if allowed by the owner. 

Any other file with "significant" changes should have the copyright owner list extended with the modification years.

Comment (scpeters) [inline urdf_model/include/urdf_model/utils.h]: Do you have any specific recommendations in this case?

Comment (scpeters) [inline urdf_model/include/urdf_model/utils.h]: Should we list OSRF as the copyright holder instead of Willow Garage?

**Comment by scpeters** [comment]
Related to https://github.com/ros/urdfdom/issues/62

File: urdf_model/include/urdf_model/utils.h (hunk 1 above)
Comment (tfoote) [inline urdf_model/include/urdf_model/utils.h]: Yes.

Comment (gerkey) [inline urdf_model/include/urdf_model/utils.h]: The general rule is that, when making non-trivial changes to existing code, you add a copyright line with appropriate owner and year. Never remove a copyright line from existing code.

But if it's new code, just create a new copyright line with appropriate content (e.g., OSRF, 2016).

Comment (scpeters) [inline urdf_model/include/urdf_model/utils.h]: Ok, I replaced Willow Garage with OSRF in b14f20e47033338f2b7c4dca5c9a978ae8a521a1. Is that ok?

Comment (scpeters) [inline urdf_model/include/urdf_model/utils.h]: Clarification: I only changed it for the new file

Comment (tfoote) [inline urdf_model/include/urdf_model/utils.h]: lgtm

**Comment by scpeters** [comment]
There were conflicts, so I rebased and squashed the license change into 5132f2a

**Comment by scpeters** [comment]
Based on the discussion in https://github.com/ros/resource_retriever/pull/7, I'm going to remove 3b71a399ae9aa349b0d9dfdbbf01bf5bad5a0940 so that the c++11 compiler flags aren't exported.

**Comment by j-rivero** [comment]
+1, nice!

**Pull request merged**